lua-plugins/cli_filter.lua.in

```diff
@@ -123,6 +123,52 @@ local function get_default_partition_or_env()
     return os.getenv('SLURM_JOB_PARTITION') or get_default_partition()
 end
 
+-- Parse slurm strings of the form 'key=value,key=value,...' to table.
+-- Supply e.g. ':' as second argument to parse strings of the form 'key:value,...' etc.
+
+local function parse_csv_tbl(csv, key_sep)
+    key_sep = key_sep or '='
+    local tbl = {}
+    for _, field in ipairs(tokenize(csv, ',')) do
+        local k, v = table.unpack(tokenize(field, key_sep, 2))
+        if v == nil then table.insert(tbl, k)
+        else tbl[k] = v
+        end
+    end
+    return tbl
+end
+
+function pp(t, i)
+    i = i or ''
+    if type(t) ~= 'table' then return tostring(t) end
+    local s =  '{'
+    local j = 0
+    i = i .. ' '
+    for k,v in pairs(t) do
+        if type(k) ~= 'number' and tonumber(k) then k = '"' .. k .. '"' end
+        if j>0 then s = s .. '\n' .. i end
+        j = j + 1
+        s = s .. ' ' .. k .. ': ' .. pp(v, i .. string.rep(' ', string.len(k)+3))
+    end
+    if j>0 then return s .. ' }' else return s .. '}' end
+end
+
+local function collect_csv_tbl(tbl, key_sep)
+    key_sep = key_sep or '='
```

Comment:
should we have this as a local variable?

Reply:
As a function argument, I'm pretty sure it is already locally bound.

Reply:
Just confirmed; this does not need to be local and doesn't modify the argument from the caller's perspective.